File: quiz_app_flutter/lib/presentation/result/ui/result_item_row.dart
import 'package:flutter/material.dart';
import 'package:quiz_app_flutter/domain/entities/result/result_data.dart';

class ResultItemRow extends StatelessWidget {
  final ResultItem item;

  const ResultItemRow({super.key, required this.item});

  @override
  Widget build(BuildContext context) {
    return Card(
      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)), // AppCardDefaults.shape equivalent
      elevation: 4, // AppCardDefaults.elevation equivalent
      child: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              item.question,
              style: const TextStyle(fontWeight: FontWeight.bold),
            ),
            const SizedBox(height: 8), // SpacerSmallHeight equivalent
            Text(
              'Answer', // Replace with string resource if using localization
              style: const TextStyle(fontWeight: FontWeight.w600),
            ),
            Padding(
              padding: const EdgeInsets.only(left: 8.0, top: 2.0),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    '• ${item.correctAnswer}',
                    style: TextStyle(
                      color: item.result ? Colors.green : Colors.red,
                    ),
                  ),
                ],
              ),
            ),
            if (item.explanation.isNotEmpty) ...[
              const SizedBox(height: 8), // SpacerSmallHeight equivalent
              Text(
                item.explanation,
                style: const TextStyle(fontStyle: FontStyle.italic),
              ),
            ],
          ],
        ),
      ),
    );
  }
}

File: quiz_app_kmp/composeApp/src/commonMain/kotlin/com/tnm/quizappkmp/presentation/result/ui/ResultItemRow.kt
package com.tnm.quizappkmp.presentation.result.ui

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Card
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.tnm.quizappkmp.domain.model.result.ResultData
import com.tnm.quizappkmp.presentation.utils.ui.AppCardDefaults
import com.tnm.quizappkmp.presentation.utils.ui.AppColors
import com.tnm.quizappkmp.presentation.utils.ui.SpacerSmallHeight
import org.jetbrains.compose.resources.stringResource
import quiz_master_kmp.composeapp.generated.resources.Res
import quiz_master_kmp.composeapp.generated.resources.label_answer


@Composable
fun ResultItemRow(item: ResultData.Item) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = AppCardDefaults.shape,
        elevation = AppCardDefaults.elevation()
    ) {
        Column(modifier = Modifier.padding(16.dp)) {
            Text(text = item.question, fontWeight = FontWeight.Bold)
            SpacerSmallHeight()
            Text(
                text = stringResource(Res.string.label_answer),
                fontWeight = FontWeight.SemiBold,
                modifier = Modifier.padding(top = 4.dp)
            )
            Column(modifier = Modifier.padding(start = 8.dp, top = 2.dp)) {
                item.correctAnswer.forEach { answer ->
                    Text(
                        text = "• $answer",
                        color = if (item.result) AppColors.SuccessColor else Color.Red
                    )
                }
            }

            if (item.explanation.isNotEmpty()) {
                SpacerSmallHeight()
                Text(text = item.explanation, fontStyle = FontStyle.Italic)
            }
        }
    }
}